Exercises/SoerenHerrmann/exercises_07.py

```diff
@@ -0,0 +1,353 @@
+# * I hope you can forgive me for this.
+
+class triplet():
+    def __init__(self, x: int, y: int, info: int) -> None:
+        self.x = int(x) 
+        self.y = int(y)
+        self.info = info
+
+
+placeholder = []
+score = 0
+
+# Initialize the plot
+fig, ax = plt.subplots()
+ax.set_aspect('equal')
+ax.set_xlim(43, 0)
+ax.set_ylim(30, 0)
+
+# Define colors for each tile type
+colors = {
+    tiles.EMPTY: 'black',
+    tiles.WALL: 'blue',
+    tiles.BLOCK: 'red',
+    tiles.PADDLE: 'green',
+    tiles.BALL: 'red'
+}
+
+
+def draw_tile(trplt: triplet):
+    global score
+    print(trplt.x, trplt.y, trplt.info)
+    if trplt.x == -1 and trplt.y == 0:
+        score = trplt.info
+        return
+    rect = plt.Rectangle((trplt.x, trplt.y), 1, 1, color=colors[tiles(trplt.info)])
+    ax.set_title(f'Score: {score}')
+    ax.add_patch(rect)
+    plt.draw()
+    plt.pause(0.00001)
+
+
+def get_element(mode: op_mode, dct: dict, idx: int, offset: int) -> int:
+    """returns element based upon the task mode"""
+    match mode:
+        case op_mode.POSITION:
+            return dct.get(dct.get(idx, 0), 0)
+        case op_mode.IMMEDIATE:
+            return dct.get(idx, 0)
+        case op_mode.RELATIVE:
+            return dct.get(dct.get(idx, 0) + offset, 0)
+
+
+def calc(dct: dict, idx: int, op: operations, instruction: list, offset: int) -> int:
+    """calc is short for calculator. Handles 1,2,7 and 8"""
+    param_one_mode = op_mode(int(instruction[0]))
+    param_two_mode = op_mode(int(instruction[1]))
+    idx += 1 
+    param_one = get_element(param_one_mode, dct, idx, offset)
+    idx += 1
+    param_two = get_element(param_two_mode, dct, idx, offset)
+    idx += 1
+    write_mode = op_mode(int(instruction[2]))
+    destination = dct.get(idx, 0) + (offset if write_mode == op_mode.RELATIVE else 0)
+    match op:
+        case operations.ADD:
+            dct[destination] = param_one + param_two
+        case operations.MULTIPLY:
+            dct[destination] = param_one * param_two
+        case operations.LESS_THAN:
+            dct[destination] = 1 if param_one < param_two else 0
+        case operations.EQUALS:
+            dct[destination] = 1 if param_one == param_two else 0
+    logging.debug(f'Calculation: {dct[destination]}')
+    return idx + 1
+
+
+def jump_operation(dct: dict,
+                   idx: int,
+                   op: operations,
+                   instruction: list,
+                   offset: int) -> int:
+    """handles the jump operations 5 and 6"""
+    param_one_mode = op_mode(int(instruction[0]))
+    param_two_mode = op_mode(int(instruction[1]))
+
+    idx += 1
+    param_one = get_element(param_one_mode, dct, idx, offset)
+    idx += 1
+    param_two = get_element(param_two_mode, dct, idx, offset)
+
+    if (op == operations.JUMP_IF_TRUE and param_one != 0) or (op == operations.JUMP_IF_FALSE and param_one == 0):
+        return param_two
+    return idx + 1
```

Comment:
idx += 1
This is redundant; please adjust it. My suggestion would be:
```suggestion
    param_one = get_element(param_one_mode, dct, idx + 1, offset)
    param_two = get_element(param_two_mode, dct, idx + 2, offset)

    if (op == operations.JUMP_IF_TRUE and param_one != 0) or (op == operations.JUMP_IF_FALSE and param_one == 0):
        return param_two
    return idx + 3
```